pkg/storage/factory.go

```diff
@@ -331,21 +374,86 @@ func (c *ClientMetrics) Unregister() {
 
 // NewObjectClient makes a new StorageClient of the desired types.
 func NewObjectClient(name string, cfg Config, clientMetrics ClientMetrics) (client.ObjectClient, error) {
-	switch name {
+	var (
+		namedStore string
+		storeType  = name
+	)
+
+	// extact storeType and lookup key for named stores
+	if words := strings.SplitN(name, ".", 2); len(words) == 2 {
+		storeType, namedStore = words[0], words[1]
+	}
```

Comment:
`NewObjectClient` has references from other packages, so I am thinking to use the same contract as before, otherwise the callers have to perform the split to make the call.